### PR TITLE
[5.8] Don't execute query when ids are empty Arrayable

### PR DESCRIPTION
**Pull request by mnabialek**
Previous condition was met only if `$ids` were empty array. If they were empty instance of `Arrayable` (for example collection with no elements) condition was false and query was executed.

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Builder.php [hunk 1]
```diff
@@ -338,7 +338,11 @@ public function find($id, $columns = ['*'])
      */
     public function findMany($ids, $columns = ['*'])
     {
-        if (empty($ids)) {
+        $empty = $ids instanceof Arrayable
+            ? empty($ids->toArray())
+            : empty($ids);
+
+        if ($empty) {
```
Comment (driesvints) [inline src/Illuminate/Database/Eloquent/Builder.php]: It's better if you cast the ids and then do the empty check in the if statement.

```php
	        $ids = $ids instanceof Arrayable ? $ids->toArray()) : $ids;

	        if (empty($ids)) {
```

That way it's also only two lines you're adding.

**Review by driesvints — CHANGES_REQUESTED** [review]

File: src/Illuminate/Database/Eloquent/Builder.php (hunk 1 above)
Comment (mnabialek) [reply src/Illuminate/Database/Eloquent/Builder.php]: It's done. I didn't want before to change argument value passed to further calls.

File: tests/Database/DatabaseEloquentBuilderTest.php [hunk 2]
```diff
@@ -117,7 +151,7 @@ public function testFindWithManyUsingCollection()
     {
         $ids = collect([1, 2]);
         $builder = m::mock(Builder::class.'[get]', [$this->getMockQueryBuilder()]);
-        $builder->getQuery()->shouldReceive('whereIn')->once()->with('foo_table.foo', $ids);
+        $builder->getQuery()->shouldReceive('whereIn')->once()->with('foo_table.foo', [1, 2]);
```
Comment (driesvints) [inline tests/Database/DatabaseEloquentBuilderTest.php]: Why did you change this? This should stay in place.

```suggestion
        $builder->getQuery()->shouldReceive('whereIn')->once()->with('foo_table.foo', $ids);
```

**Review by driesvints — DISMISSED** [review]

File: tests/Database/DatabaseEloquentBuilderTest.php (hunk 2 above)
Comment (mnabialek) [reply tests/Database/DatabaseEloquentBuilderTest.php]: This test started to fail because now we don't pass collection any more, just an array because we made a cast

Comment (driesvints) [reply tests/Database/DatabaseEloquentBuilderTest.php]: The test is meant to proof that a collection can also pass so we need to make sure this still succeeds. Otherwise it's a breaking change.

Comment (mnabialek) [reply tests/Database/DatabaseEloquentBuilderTest.php]: Well, but collection is still passed in this test, we just need to modify expectation because now into `whereIn` array is passed because now as suggested at the beginning of findMany we do:
```
$ids = $ids instanceof Arrayable ? $ids->toArray() : $ids;
```
so we pass into `whereKey` array and not collection any more and further into `whereIn` now array is passed and not collection any more.

Comment (driesvints) [reply tests/Database/DatabaseEloquentBuilderTest.php]: Right! Thanks for point that out :)
